api/src/main/java/net/kyori/adventure/audience/Audience.java

```diff
@@ -608,6 +611,25 @@ default void showBossBar(final @NotNull BossBar bar) {
   default void hideBossBar(final @NotNull BossBar bar) {
   }
 
+  /**
+   * Check if this audience has been shown this BossBar
+   * @param bar BossBar
+   * @return boolean
+   * @since 4.12.0
+   */
+  default boolean hasBossBar(final @NotNull BossBar bar) {
+    return false;
+  }
+
+  /**
+   * Returns a set containing all BossBars this Audience has been shown
+   * @return An unmodifiable set of BossBars
+   * @since 4.12.0
+   */
+  default @NotNull @UnmodifiableView Set<BossBar> getBossBars() {
```

Comment:
1. This does not follow Kyori's organizational naming standards -- you shouldn't have `get` prefixes.
2. We can't really get *all* boss bars here, especially if we're in a proxy environment.